main.go

```diff
@@ -40,16 +45,27 @@ func main() {
 		log.Fatalln(err)
 	}
 
-	// retrieve raw from graphql
+	var sleepDur time.Duration
+	if !runOnce {
+		sleepDur, err = time.ParseDuration(envVars["RECONCILE_SLEEP_TIME"])
+		if err != nil {
+			log.Fatalln(err)
+		}
 
-	sleepDur, err := time.ParseDuration(envVars["RECONCILE_SLEEP_TIME"])
-	if err != nil {
-		log.Fatalln(err)
+		// configure prometheus metrics handler
+		http.Handle("/metrics", promhttp.Handler())
+		go func() {
+			http.ListenAndServe(fmt.Sprintf(":%s", envVars["METRICS_SERVER_PORT"]), nil)
+		}()
 	}
 
+	// retrieve raw from graphql
+
 	for {
-		ctx := context.Background()
+		status := 0
+		start := time.Now()
 
+		ctx := context.Background()
```

Comment:
Tell me more about this

Reply:
on each iteration (reconcile loop), a new context is created. This is later used to [derive a timeout context](https://github.com/app-sre/git-partition-sync-producer/blob/master/pkg/uploader.go#L212) when querying the graphql server

Reply:
Actually I see its not a change.

pass the context into the http handlers and have a way to exit gofuncs nested in handlers

status/start needed for metrics

right?

Reply:
correct. status/start for metrics. main ctx part is currently for setting deadline on graphql query. Certainly room for further refactors that utilize ctx